src/DeleteComment.jsx

```diff
@@ -0,0 +1,26 @@
+import { useContext, useEffect, useState } from "react";
+import { deleteComment } from "./apis";
+import { UserContext } from "./User";
+
+export const DeleteComment = ({author, comment_id, setListOfComments, listOfComments, setIsLoading}) => {
+    const { user, isLoggedIn } = useContext(UserContext)
+
+    const deleteCommentById = () => {
+        const commentListDeleted = listOfComments.filter((comment) => {return comment.comment_id !== comment_id})
+        const commentList = [...listOfComments]
+        setIsLoading(true)
+        setListOfComments(commentListDeleted)
+        deleteComment(comment_id).then(() => {
+            setIsLoading(false)
+            alert("Comment Deleted")
+        }).catch(() => {
+            alert("sorry something went wrong, your comment was not deleted, please try again later.")
+            setListOfComments(commentList)
+        })
+    }
+
+    if (isLoggedIn && user.username === author){
```

Comment:
This is a great way of only rendering the delete button if the user matches the logged in user 👍 

Currently when I try to delete a comment I've just added I get an error message (down to the temporary ID we're giving with optimistic rendering) you could add in another condition here to stop the delete rendering if the ID doesn't look like a real one to prevent this.